packages/sdk/src/subgraph/subgraph.ts

```diff
@@ -190,6 +191,16 @@ export class Subgraph {
   async getActiveTransactions(_requestContext?: RequestContext): Promise<ActiveTransaction[]> {
     const { requestContext, methodContext } = createLoggingContext(this.getActiveTransactions.name, _requestContext);
 
+    // For each chain, update subgraph sync status
+    await this.updateSyncStatus();
+    const inSyncChains: number[] = Object.entries(this.sdks).filter(([, sdk]) => sdk.inSync).map(([chainId]) => parseInt(chainId));
+    if (inSyncChains.length === 0) {
+      // If no chains are in-sync, return an empty array.
+      // TODO: Or throw an error - all chains out of sync?
+      this.logger.warn("Cannot get active transactions; all subgraphs for all chains are out of sync!", requestContext, methodContext);
+      return [];
+    }
```

Comment:
active transactions should return as many as possible transactions, even if the subgraph is out of sync. similar to historical transactions

Reply:
cc @rhlsthrm want to make sure this is desired behavior. seems like this is pretty risk-free in the sdk but in the router could lead to some undesired actions on subgraph loops

Reply:
This would result in potential duplicates, though, which means wasted gas.
Ex.
getActiveTransactions -> new sender prepare (at block N) -> handle, add to our local active txs -> handled -> clear from local active txs.
subg goes out of sync, stuck at block N -> should be considered out of sync but we ignore it -> get the same sender prepare as before (at block N) -> try to handle -> P019

but maybe we don't care about these edge cases too much..?

Reply:
Yes I agree with @LayneHaber I think we need to return all active transactions even if it's out of sync.

> subg goes out of sync, stuck at block N -> should be considered out of sync but we ignore it -> get the same sender prepare as before (at block N) -> try to handle -> P019

This flow does not happen on the SDK, the SDK doesn't handle prepares by preparing. On the subgraph it will just show up as active.

Reply:
Okay, I'm going to make it the default for the `request` method that sync is *not* required